packages/commons-test/test/riskAnalysisValidation.integration.test.ts

```diff
@@ -0,0 +1,715 @@
+      multiAnswers: [],
+    };
+
+    const result = validateRiskAnalysis(
+      validSchemaOnlyRiskAnalysis2_0_Private,
+      true,
+      "PRIVATE"
+    );
+
+    expect(result).toEqual({
+      type: "valid",
+      value: expected,
+    });
+  });
+
+  it("should succeed on correct form 2.0 on tenant kind GSP", () => {
+    const result = validateRiskAnalysis(
+      validRiskAnalysis2_0_Private,
+      false,
+      "GSP"
+    );
+    const resultSchemaOnly = validateRiskAnalysis(
+      validRiskAnalysis2_0_Private,
+      true,
+      "GSP"
+    );
+
+    expect(result).toEqual({
+      type: "valid",
+      value: expectedValidatedRiskANalysis2_0_Private,
+    });
+    expect(result).toEqual(resultSchemaOnly);
+  });
+
+  it("should succeed on correct form 2.0 schema only on tenant kind GSP", () => {
+    const expected: RiskAnalysisValidatedForm = {
+      version: validSchemaOnlyRiskAnalysis2_0_Private.version,
+      singleAnswers: [{ key: "purpose", value: "INSTITUTIONAL" }],
+      multiAnswers: [],
+    };
+
+    const result = validateRiskAnalysis(
+      validSchemaOnlyRiskAnalysis2_0_Private,
+      true,
+      "GSP"
+    );
+
+    expect(result).toEqual({
+      type: "valid",
+      value: expected,
+    });
+  });
+
+  it("should fail if version does not exists", () => {
+    const invalidRiskAnalysis: RiskAnalysisFormToValidate = {
+      ...validRiskAnalysis3_0_Pa,
+      version: "9999.0",
+    };
+
+    expect(validateRiskAnalysis(invalidRiskAnalysis, false, "PA")).toEqual({
+      type: "invalid",
+      issues: [unexpectedRulesVersionError(invalidRiskAnalysis.version)],
+    });
+
+    const invalidRiskAnalysis2: RiskAnalysisFormToValidate = {
+      ...validRiskAnalysis2_0_Private,
+      version: "not a valid version",
+    };
+
+    expect(
+      validateRiskAnalysis(invalidRiskAnalysis2, false, "PRIVATE")
+    ).toEqual({
+      type: "invalid",
+      issues: [unexpectedRulesVersionError(invalidRiskAnalysis2.version)],
+    });
+  });
+
+  it("fail if a provided answer depends on missing fields", () => {
+    const riskAnalysisPa: RiskAnalysisFormToValidate = {
+      version: validRiskAnalysis3_0_Pa.version,
+      answers: {
+        ...validRiskAnalysis3_0_Pa.answers,
+        doneDpia: [],
+        confirmedDoneDpia: ["YES"], // confirmedDoneDpia requires doneDpia to be present
+        policyProvided: [],
+        reasonPolicyNotProvided: ["reason"], // reasonPolicyNotProvided requires policyProvided to be present
+      },
+    };
+
+    expect(validateRiskAnalysis(riskAnalysisPa, false, "PA")).toEqual({
+      type: "invalid",
+      issues: [
+        dependencyNotFoundError("reasonPolicyNotProvided", "policyProvided"),
+        dependencyNotFoundError("confirmedDoneDpia", "doneDpia"),
+        missingExpectedFieldError("policyProvided"),
+        missingExpectedFieldError("doneDpia"),
+      ],
+    });
+
+    const riskAnalysisPrivate: RiskAnalysisFormToValidate = {
```

Comment:
IMO we can avoid testing different versions in this case because we are checking a general purpose feature of the validator, not a specific version.
Pointing this out just because it could became complicated to maintain and it provides minimal additional coverage